File: Maths_And_Stats/Number_Theory/Segmented_Sieve/segmented_sieve.py
def segmented_sieve(n):

    # Create an boolean array with all values True
    primes = [True]*n

    for p in range(2,n):
        #If prime[p] is True,it is a prime and its multiples are not prime

        if primes[p]:
            for i in range(2*p,n,p):

                # Mark every multiple of a prime as not prime
                primes[i]=False

    #If value is true it is prime and print value
    for l in range(2,n):
        if primes[l]:
            print(f"{l} ")

#Test
while True:
    try:
        input_value = int(input("Please a number: "))
        segmented_sieve(input_value)
        break
    except ValueError:
        print("No valid integer! Please try again ...")

File: Maths/Number-Theory/Segmented_Sieve/segmented_sieve.js
function segmented_sieve(n){
    //Create an boolean array with all values True
    var primes = new Array(n).fill(true)

    for(var i=2;i<=n;i++){
        //If prime[p] is True,it is a prime and its multiples are not prime
        if(primes[i]){
            for(var j=i*2;j<=n;j+=i){

                //Mark every multiple of a prime as not prime
                primes[j]=false
            }
        }
    }
    //If value is true it is prime and print value
    for(var k=2;k<n;k++){
        if(primes[k])
            console.log(k)
    }
}

//Test
let number = prompt("Enter a positive integer")
segmented_sieve(parseInt(number))